packages/fast-usdc-contract/test/supports.ts

```diff
@@ -189,55 +194,104 @@ export const commonSetup = async (t: ExecutionContext<any>) => {
 
   await registerKnownChains(agoricNamesAdmin, () => {});
 
-  // XXX: copied from orchestration supports
-  /** proxy for current sequence of the IBC Channel */
-  let ibcSequenceNonce = 0n;
   /**
-   * Simulate an inbound message to the vtransfer bridge.
-   *
-   * Uses the localchain bridge to lookup the last message and infer packet
-   * and transaction details.
+   * Find the Nth outgoing MsgTransfer and its sequence number from the localchain bridge log.
+   * @param index 0-based index from the start, or negative index from the end.
+   * @returns The MsgTransfer message and its sequence number.
+   * @throws If index is out of bounds or sequence is not found in the log result.
+   */
+  const getOutgoingTransfer = (index: number) => {
+    const log = inspectLocalBridgeLog();
+    const transferMessagesInfo: { message: MsgTransfer; sequence: bigint }[] =
+      [];
+
+    for (const entry of log) {
+      const { obj, result } = entry;
+      if (
+        obj.type === 'VLOCALCHAIN_EXECUTE_TX' &&
+        obj.messages &&
+        Array.isArray(result)
+      ) {
+        for (let i = 0; i < obj.messages.length; i += 1) {
+          const message = obj.messages[i];
+          if (
+            message['@type'] === '/ibc.applications.transfer.v1.MsgTransfer'
+          ) {
+            const msgResult = result[i];
+            // Check sequence exists in result object associated with this message
+            if (msgResult && typeof msgResult.sequence !== 'undefined') {
+              transferMessagesInfo.push({
+                message,
+                sequence: BigInt(msgResult.sequence),
+              });
+            } else {
+              throw new Error(
+                `Sequence not found in result for MsgTransfer: ${JSON.stringify(message)}`,
+              );
+            }
+          }
+        }
+      }
+    }
+
+    const totalTransfers = transferMessagesInfo.length;
+    let targetIndex = index;
+    if (index < 0) {
+      targetIndex = totalTransfers + index;
+    }
+
+    if (targetIndex < 0 || targetIndex >= totalTransfers) {
+      throw new Error(
+        `Index ${index} out of bounds for ${totalTransfers} outgoing transfers.`,
+      );
+    }
+
+    return transferMessagesInfo[targetIndex];
+  };
+
+  /**
+   * Simulate an inbound VTransferIBCEvent message to the vtransfer bridge for a specific transfer.
    *
-   * Tracks `sequence` locally in test context, so this helper must be used
-   * for all simulated VTransfer calls in a test run for sequence to be
-   * accurate.
+   * @param event The type of IBC event ('acknowledgementPacket' or 'timeoutPacket').
+   * @param transferId Index of the outgoing transfer (e.g., -1 for the last one) or the `{ message: MsgTransfer, sequence: bigint }` object itself.
```

Comment:
It might be clearer to call this something other than a `transferId`.  Maybe `selectedTransfer` or `query` or something?